sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -445,6 +445,26 @@ class Analyzer(
         val newOrdering = resolveSortOrders(ordering, child, throws = false)
         Sort(newOrdering, global, child)
 
+     // Resolve the order index to be a specific column
+      case s @ Sort(ordering, global, child) if child.resolved && s.resolved =>
+        def indexToColumn(index: Int, direction: SortDirection) = {
+          val orderNodes = child.output
+          if (index > 0 && index <= orderNodes.size) {
+            SortOrder(orderNodes(index - 1), direction)
+          } else {
+            throw new UnresolvedException(s,
+              s"""Order by position: $index does not exist \n
+                  |The Select List is indexed from 1 to ${orderNodes.size}""".stripMargin)
+          }
+        }
+        val newOrdering = ordering map {
```

Comment:
oh, actually, I meant if we can just check if there is any integer literal. If so, we create a new `Sort`. Otherwise, we keep the old one.